torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3177,12 +3177,14 @@ def reference_lgamma(x):
                    dtypesIfCUDA=floating_types_and(torch.float16),
                    assert_autodiffed=True),
     UnaryUfuncInfo('exp2',
+                   aliases=('special.exp2', ),
                    ref=np_unary_ufunc_integer_promotion_wrapper(np.exp2),
```

Comment:
Have kept `np.exp2` as reference as it matches Pytorch promotion behavior for floats (similar for expm1).
`scipy.special.exp2` promotes float16 to float32 (thus there is a mismatch in result)

```python
>>> import torch
>>> import numpy as np
>>> import scipy.special
>>> import math
>>> t = torch.tensor([math.pi/2], dtype=torch.float16)
>>> torch.exp2(t)
tensor([2.9688], dtype=torch.float16)
>>> np.exp2(t.numpy())
array([2.969], dtype=float16)
>>> scipy.special.exp2(t.numpy())
array([2.9696903], dtype=float32)
```